Client/Frontend/Share/ShareExtensionHelper.swift

```diff
@@ -54,27 +66,69 @@ class ShareExtensionHelper: NSObject {
         // which is after the user taps the button. So a million cycles away.
         findLoginExtensionItem()
 
-        activityViewController.completionWithItemsHandler = { activityType, completed, returnedItems, activityError in
-            if !completed {
-                completionHandler(completed, activityType)
-                return
-            }
-            // Bug 1392418 - When copying a url using the share extension there are 2 urls in the pasteboard.
-            // This is a iOS 11.0 bug. Fixed in 11.2
-            if UIPasteboard.general.hasURLs, let url = UIPasteboard.general.urls?.first {
-                UIPasteboard.general.urls = [url]
-            }
+        activityViewController.completionWithItemsHandler = { [weak self] activityType, completed, returnedItems, activityError in
+            guard let self = self else { return }
             
-            if self.isPasswordManagerActivityType(activityType.map { $0.rawValue }) {
-                if let logins = returnedItems {
-                    self.fillPasswords(logins as [AnyObject])
+            if self.shareActivityType(activityType.map { $0.rawValue }) != .iBooks { //!self.isOpenInIBooksActivityType(activityType.map { $0.rawValue }) {
+                if !completed {
+                    completionHandler(completed, activityType, nil)
+                    return
+                }
+                // Bug 1392418 - When copying a url using the share extension there are 2 urls in the pasteboard.
+                // This is a iOS 11.0 bug. Fixed in 11.2
+                if UIPasteboard.general.hasURLs, let url = UIPasteboard.general.urls?.first {
+                    UIPasteboard.general.urls = [url]
+                }
+                
+                if self.shareActivityType(activityType.map { $0.rawValue }) != .password { //self.isPasswordManagerActivityType(activityType.map { $0.rawValue }) {
+                    if let logins = returnedItems {
+                        self.fillPasswords(logins as [AnyObject])
+                    }
                 }
-            }
 
-            completionHandler(completed, activityType)
+                completionHandler(completed, activityType, nil)
+            } else {
+                self.writeWebPagePDFDataToURL { url, error in
```

Comment:
same with .mail right?

Reply:
No this is actually sharing the website itself so in mail sharing, it needs to share the link directly which it does right now. However if it is a url that opens a pdf in browser, it is creating a temporary document and attaches it to mail.